### DISCUSSION TITLE
Missing node.js core modules following upgrade webpack-encore 0 to 1

### DISCUSSION THREAD

**Discussion by neilf184** [symfony/symfony#47722] [Q&A]
I'm attempting to upgrade weback-encore from version 0.31.1 to 1.8.2

The problem I'm having is that my application won't build as none of the node.js core modules are found:-

"./node_modules/buffer-equal-constant-time/index.js" contains a reference to the file "buffer".
This file can not be found, please check it for typos or update it if the file got moved.

"./node_modules/jwa/index.js" contains a reference to the file "crypto".
This file can not be found, please check it for typos or update it if the file got moved.

"./node_modules/jwa/index.js" contains a reference to the file "util".
This file can not be found, please check it for typos or update it if the file got moved.

"./node_modules/jws/lib/data-stream.js" contains a reference to the file "stream".
This file can not be found, please check it for typos or update it if the file got moved.

My understanding is that this in because webpack 5 doesn't define polyfills that were present in webpack 4, but I've been unable to find anything that tells me what I need to do to get this working in Symfony webpack-encore

**Comment by stof** [comment]
webpack 5 has disabled the node polyfills by default. See https://www.npmjs.com/package/node-polyfill-webpack-plugin for a plugin adding them back

**Reply by stof** [reply]
Alternatively, you could also migrate from [`jws`](https://www.npmjs.com/package/jws) to [`jose`](https://www.npmjs.com/package/jose) which has native support for web browsers.

**Reply by neilf184** [reply]
Thanks for the very quick reply! I've seen the node-polyfill-webpack-plugin page but I don't understand how to add this into my webpack-encore webpack.config.js file.

**Reply by stof** [reply]
by using the dedicated configuration method allowing to add custom plugins, which is named `addPlugin`: https://symfony.com/doc/current/frontend/encore/custom-loaders-plugins.html#adding-custom-plugins

**Reply by neilf184** [reply]
Many thanks, that's resolved that particular problem
